src/commands/install/install_command.py

```diff
@@ -0,0 +1,81 @@
+from logging import getLogger
+from os import getcwd
+from typing import Any, List, Optional
+
+from src.commands.install.install_package_from_repo import install_package_from_repo
+from src.commands.install.pull_package_submodules import pull_package_submodules
+from src.cli import Command
+from src.utils import Project, extract_info_from_repo_id, log_color_provider
+
+EXTERNAL_DEPENDENCY_REFERENCE_DESCRIPTION = """- `GITHUB_ACCOUNT_NAME/REPO_NAME[@TAG]`
+    - `OpenZeppelin/cairo-contracts@0.1.0`
+- `URL_TO_THE_REPOSITORY`
+    - `https://github.com/OpenZeppelin/cairo-contracts`
+- `SSH_URI`
+    - `git@github.com:OpenZeppelin/cairo-contracts.git`
+"""
+
+
+class InstallCommand(Command):
+    def __init__(self, project: Project) -> None:
+        super().__init__()
+        self._project = project
+
+    @property
+    def name(self) -> str:
+        return "install"
+
+    @property
+    def description(self) -> str:
+        return "Install a dependency as a git submodule."
+
+    @property
+    def example(self) -> Optional[str]:
+        return "$ protostar install https://github.com/OpenZeppelin/cairo-contracts"
+
+    @property
+    def arguments(self) -> List[Command.Argument]:
+        return [
+            Command.Argument(
+                name="package",
+                description=EXTERNAL_DEPENDENCY_REFERENCE_DESCRIPTION,
+                type="str",
+                is_positional=True,
+            ),
+            Command.Argument(
+                name="name",
+                description="A custom package name. Use it to resolve name conflicts.",
+                type="str",
+            ),
+        ]
+
+    async def run(self, args):
+        handle_install_command(args, self._project)
+
+
+def handle_install_command(args: Any, project: Project) -> None:
```

Comment:
i don't see a reason why it'd be defined in a separate method

Reply:
Avoiding changing more code in this PR. I'd have to update the code and tests for every function otherwise.

Reply:
Please create a task for inlining this methods then

Reply:
https://github.com/software-mansion/protostar/issues/241